__tests__/unit/core-manager/server/server.test.ts

```diff
@@ -224,4 +233,105 @@ describe("Server", () => {
+            pluginsConfiguration.basicAuthentication.users = [
+                {
+                    username: "username",
+                    password: "$argon2id$v=19$m=4096,t=3,p=1$NiGA5Cy5vFWTxhBaZMG/3Q$TwEFlzTuIB0fDy+qozEas+GzEiBcLRkm5F+/ClVRCDY"
+                }
+            ];
+
+            injectOptions = {
+                method: "POST",
+                url: "/",
+                payload: {
+                    jsonrpc: "2.0",
+                    id: "1",
+                    method: "dummy",
+                    params: { id: 123 },
+                },
+                headers: {
+                    "content-type": "application/vnd.api+json",
+                },
+            };
+        })
+
+        it("should return RCP error if no username and password in header", async () => {
+            await server.initialize("serverName", {})
+            await server.boot()
+
+            const response = await server.inject(injectOptions);
+            const parsedResponse: Record<string, any> = { body: response.result, statusCode: response.statusCode };
+
+            expect(parsedResponse).toEqual({
+                body: {
+                    jsonrpc: '2.0',
+                    error: { code: -32001, message: 'Missing authentication' },
+                    id: null
+                },
+                statusCode: 200
+            })
+        });
+
+        it("should return RCP error if password is invalid", async () => {
+            await server.initialize("serverName", {})
+            await server.boot()
+
+            // Data in header: { username: "username", password: "invalid" }
+            injectOptions.headers.Authorization = "Basic dXNlcm5hbWU6aW52YWxpZA==";
+
+            const response = await server.inject(injectOptions);
+            const parsedResponse: Record<string, any> = { body: response.result, statusCode: response.statusCode };
+
+            expect(parsedResponse).toEqual({
+                body: {
+                    jsonrpc: '2.0',
+                    error: { code: -32001, message: 'Bad username or password' },
+                    id: null
+                },
+                statusCode: 200
+            })
+        });
+
+        it("should return RCP error user not found", async () => {
+            pluginsConfiguration.basicAuthentication.users = [];
+
+            await server.initialize("serverName", {})
+            await server.boot()
+
+            // Data in header: { username: "username", password: "password" }
+            injectOptions.headers.Authorization = "Basic dXNlcm5hbWU6cGFzc3dvcmQ=";
+
+            const response = await server.inject(injectOptions);
+            const parsedResponse: Record<string, any> = { body: response.result, statusCode: response.statusCode };
+
+            expect(parsedResponse).toEqual({
+                body: {
+                    jsonrpc: '2.0',
+                    error: { code: -32001, message: 'Bad username or password' },
```

Comment:
```suggestion
                    error: { code: -32001, message: 'These credentials do not match our records.' },
```

Reply:
This was an error message from hapi/basic plugin. Now all 401 errors have new message.